wp-content/plugins/tapestry/controller/class.tapestry-controller.php

```diff
@@ -55,6 +55,26 @@ public function updateTapestry($tapestry) {
         return $tapestry;
     }
 
+    /**
+     * Update Tapestry settings
+     * 
+     * @param  Object @settings
+     * @return Object Tapestry settings
+     */
+    public function updateTapestrySettings($settings) {
+        if (!$this->postId) {
+            return $this->_throwsError('INVALID_POST_ID');
+        }
+
+        $tapestry = get_post_meta($this->postId, 'tapestry', true);
+        $tapestry->settings = $settings;
```

Comment:
Could you add a todo here to add some validation for the $settings that is being passed in?

Reply:
Should it be validated in the `endpoints.php`?

Reply:
I think doing the validation here makes more sense actually.